### DISCUSSION TITLE
⌛ 🔗 Improved form widget - Action for Login, Registration And Forget password.

### DISCUSSION THREAD

**Discussion by Joliamendi** [elementor/elementor#21600] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hi elementor team

The new Elementor updates come with great features. It would be a great feature if we could create a login, registration and Forget password form using a form widget. Because this widget is very powerful in customization.

There is already a login widget, but it would be great to integrate it with the form widget.

### Describe the solution you'd like

The possibility of creating a login, registration and forget password form in the form widget.

Also some more options for it would be great.

1- Possibility of custom redirection after login or registration for different user roles.

2- WooCommerce checkout page redirect (If the user is not logged in, if she enters the checkout page, she will be automatically redirected to the login / registration page)

3- Redirect after login for checkout page (In continuation of the above feature, after the user enters, she will be redirected to the checkout page)

**I hope users will vote for this feature to be added to the Elementor roadmap. This feature is really needed.**

**Comment by daniboi21** [comment]
I really need this on my ecommerce website! Currently Elementor only has Login widget, and i have to search for other plugin to do registration form, which is not very ideal. Elementor PLEASE look into this as soon as possible.

**Comment by drewclifton** [comment]
Registration is needed! Currently we are forced to find some third party thing for this ONE fundamental need (and these bloated plugins usually come with 30-something other widgets we don't need)!

Please just add a User Registration widget, or extend the Login widget to choose between Login and Registration.

Thank you - all the best.

**Comment by joaodath** [comment]
I'd extend this feature request to add Google reCAPTCHA v3 and 2-Factor Authentication too either natively or integrating with Wordfence. 
Wordfence adds their 2FA prompt to wp-login.php form, so maybe integrating the wp-login form directly could be a way to do that. 

This is a long overdue request from the community!

References: 
[Feature Request - 2 Factor Auth Integration with Elementor Login Form #5183](https://github.com/elementor/elementor/issues/5183)
[Add reCaptcha to Login widget in Pro #3776](https://github.com/elementor/elementor/issues/3776)

**Comment by dbjpanda** [comment]
Maybe you can check Actions Pack plugin for Elementor
